db/view/view_building_worker.cc

```diff
@@ -45,7 +45,10 @@ class view_building_worker::consumer : public view_consumer {
 
 protected:
     virtual void load_views_to_build() override {
-        _views_to_build = _batch.tasks | std::views::filter([this] (const auto& task_entry) {
+        // List of tasks might be modified by aborting some of them.
+        // Make a local copy to avoid races.
+        auto tasks_in_batch = _batch.tasks;
+        _views_to_build = tasks_in_batch | std::views::filter([this] (const auto& task_entry) {
```

Comment:
Do I understand correctly that `_batch.tasks` can be modified concurrently _by another shard_, and that's why you want to copy? It won't help, copying the vector like you try to do here is not an atomic operation.

It's rather dangerous to keep long-living references to things in other shards, if those things can be mutated.

Reply:
changed to guarding with mutex